finagle-mysql/src/main/scala/com/twitter/finagle/mysql/ConnectionInitSql.scala

```diff
@@ -0,0 +1,30 @@
+package com.twitter.finagle.mysql
+
+import com.twitter.finagle._
+import com.twitter.finagle.mysql.param.ConnectionInitRequest
+import com.twitter.util._
+
+/**
+  * A stack module that executes a request when create a connection.
+  */
+object ConnectionInitSql {
+  val Role: Stack.Role = Stack.Role("ConnectionInitSql")
+
+  private[finagle] def module: Stackable[ServiceFactory[Request, Result]] =
+    new Stack.Module1[ConnectionInitRequest, ServiceFactory[Request, Result]] {
+      val role: Stack.Role = Role
+      val description: String = "Installs a connection init sql module in the stack"
+
+      def make(in: ConnectionInitRequest, next: ServiceFactory[Request, Result]): ServiceFactory[Request, Result] = {
+        next.flatMap { service =>
+          in.request.fold(Future.value(service)) { req =>
+            service(req).flatMap {
+              case _: OK => Future.value(service)
+              case Error(code, sqlState, message) =>
```

Comment:
In the error case I believe we need to make sure to close the `service`.

Reply:
add service.close in error case.
https://github.com/twitter/finagle/pull/840/commits/c0bc82e6504441cd5cc3db80ae1d4075fccc24cc

Reply:
I was wrong: the `flatMap` operation will close the underlying service if we get a `Future.exception` out of it. I'll change it back in the internal process.